File: Plugins/Aspose_Diagram_Java_for_Ruby/lib/asposediagramjava/SolutionXML/readxmlvalues.rb
module Asposediagramjava
  module ReadXmlValues
    def initialize()
        data_dir = File.dirname(File.dirname(File.dirname(File.dirname(__FILE__)))) + '/data/'

        # Create instance of Diagram
        diagram = Rjb::import('com.aspose.diagram.Diagram').new(data_dir + "SolutionXmlElement.vdx")

        solution_xmls = diagram.getSolutionXMLs()

        i = 0
        while i < solution_xmls.getCount()
            solution_xml = solution_xmls.get(i)
            # get name property
            puts "Name: " + solution_xml.getName().to_s
            # get xml value
            puts "Value:" + solution_xml.getXmlValue().to_s

            i +=1
        end
    end
  end
end

File: Plugins/Aspose_Diagram_Java_for_PHP/src/aspose/diagram/WorkingwithSolutionXMLElements/ReadXmlValues.php
<?php
namespace Aspose\Diagram\WorkingwithSolutionXMLElements;

use com\aspose\diagram\Diagram as Diagram;
use com\aspose\diagram\SolutionXML as SolutionXML;
use com\aspose\diagram\SaveFileFormat as SaveFileFormat;
class ReadXmlValues{

    public static function run($dataDir=null){

        # Create instance of Diagram
        $diagram = new Diagram($dataDir."Drawing.vdx");

        $solution_xmls=$diagram->getSolutionXMLs();

        $i= 0;
        while ($i<(int)(string)$solution_xmls->getCount()) {
        $solution_xml =$solution_xmls->get($i);
        # get name property
        print "Name: ".(string)$solution_xml->getName();
        # get xml value
        print "Value:".(string)$solution_xml->getXmlValue();
        $i += 1;
        }
    }
}